components/brave_shields/core/browser/ad_block_service_helper.cc

```diff
@@ -107,4 +90,47 @@ void MergeResourcesInto(base::Value::Dict from,
   }
 }
 
+// Removes any procedural filters from the given UrlCosmeticResources Value.
+//
+// Procedural filters are filters with at least one selector operator of a type
+// that isn't `css-selector`.
+//
+// These filters are represented as JSON provided by adblock-rust. The format
+// is documented at:
+// https://docs.rs/adblock/latest/adblock/cosmetic_filter_cache/struct.ProceduralOrActionFilter.html
+void StripProceduralFilters(base::Value::Dict& resources) {
```

Comment:
Why is this in a separate helper file? If this is just for testing, you should add a friend test so it can be a private method in the adblock service

Reply:
or is this to make it accessible to ios?

Reply:
We have two other unit-tested helper methods in there too, seemed like it fit with those.

I don't think we have a mockable version of `AdBlockService` just yet, moving it there would mean we need to use browsertests.

Reply:
I guess this is fine for now, but I'm not a fan of exposing these as public methods. Can we at least move it into an `internal` target and restrict visibility? This should ideally be the case with any classes that shouldn't be used outside the component.

Reply:
moved